tla-types/src/test/scala/at/forsyte/apalache/tla/typecomp/TestHybrid.scala

```diff
@@ -219,6 +219,221 @@ class TestHybrid extends BuilderTest {
 
   }
 
+  test("letIn") {
+    type T = TBuilderInstruction
+    type DeclParamT = (TlaType1, Seq[TlaType1])
+    type TParam1 = (TlaType1, DeclParamT)
+    type TParamN = (TlaType1, Seq[DeclParamT])
+
+    val parameterTypeGen: Gen[TlaType1] = for {
+      n <- Gen.choose(1, 5)
+      ts <- Gen.listOfN(n, tt1gen.genPrimitive)
+    } yield ts match {
+      case head :: Nil  => head
+      case head :: tail => OperT1(tail, head)
+      case Nil          => IntT1 // impossible, since 1 <= n <= 5, but the compiler doesn't know and complains
```

Comment:
appreciate the message here! I suggest throwing an exception here with the comment as the message instead of returning the type. That will ensure that if this ever does get hit for some reason we'll get the expected error rather than returning an int arbitrarily :)

Reply:
That's fair, I'll do that.